src/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,242 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+// import { createStyleSheet } from 'jss-theme-reactor';
+// import classNames from 'classnames';
+import TextField, { TextFieldInput, TextFieldLabel } from '../TextField';
+import TimePickerDialog from './TimePickerDialog';
+import { formatTime } from '../utils/timeUtils';
+
+
+const emptyTime = new Date();
+emptyTime.setHours(0);
+emptyTime.setMinutes(0);
+emptyTime.setSeconds(0);
+emptyTime.setMilliseconds(0);
+
+/* export const styleSheet = createStyleSheet('TimePicker', (theme) => {
+  return {}
+});*/
+
+class TimePicker extends Component {
+  static propTypes = {
+    /**
+     * If true, automatically accept and close the picker on set minutes.
+     */
+    autoOk: PropTypes.bool,
+    /**
+     * Override the label of the 'Cancel' button.
+     */
+    cancelLabel: PropTypes.node,
+    /**
+     * The initial time value of the TimePicker.
+     */
+    defaultTime: PropTypes.object,
```

Comment:
To begin with, I think that we should remove any uncontrolled properties like `defaultTime`. That should help with https://github.com/callemall/material-ui/issues/4719.